sql-plugin/src/main/scala/com/nvidia/spark/rapids/collectionOperations.scala

```diff
@@ -16,10 +16,74 @@
 
 package com.nvidia.spark.rapids
 
-import ai.rapids.cudf.ColumnVector
+import scala.collection.mutable.ArrayBuffer
+
+import ai.rapids.cudf.{ColumnVector, ColumnView}
+import com.nvidia.spark.rapids.RapidsPluginImplicits._
 
 import org.apache.spark.sql.catalyst.expressions.Expression
 import org.apache.spark.sql.types._
+import org.apache.spark.sql.vectorized.ColumnarBatch
+import org.apache.spark.unsafe.types.UTF8String
+
+case class GpuConcat(children: Seq[Expression]) extends GpuComplexTypeMergingExpression {
+
+  @transient override lazy val dataType: DataType = {
+    if (children.isEmpty) {
+      StringType
+    } else {
+      super.dataType
+    }
+  }
+
+  override def nullable: Boolean = children.exists(_.nullable)
+
+  override def columnarEval(batch: ColumnarBatch): Any = dataType match {
+    case StringType => stringConcat(batch)
+    case ArrayType(_, _) => listConcat(batch)
+  }
+
+  private def stringConcat(batch: ColumnarBatch): GpuColumnVector = {
+    val rows = batch.numRows()
+
+    withResource(ArrayBuffer.empty[ColumnVector]) { buffer =>
+      withResource(Scalar.fromString(null)) { nullScalar =>
+        // build input buffer
+        children.foreach { child =>
+          child.columnarEval(batch) match {
+            case cv: GpuColumnVector =>
+              buffer += cv.getBase
+            case null =>
+              buffer += GpuColumnVector.from(nullScalar, rows, StringType).getBase
+            case sv: Any =>
+              withResource(Scalar.fromString(sv.asInstanceOf[UTF8String].toString)) { scalar =>
+                buffer += GpuColumnVector.from(scalar, rows, StringType).getBase
+              }
+          }
+        }
+        // run string concatenate
+        withResource(Scalar.fromString("")) { emptyScalar =>
+          GpuColumnVector.from(ColumnVector.stringConcatenate(emptyScalar, nullScalar,
+            buffer.toArray[ColumnView]), StringType)
```

Comment:
Could you please file a bug against cudf to remove the assert for at least two entries, and an issue against ourselves to add a test to verify that we can concat just a single string column?

https://github.com/rapidsai/cudf/blob/92f067dcd80ce11c0431432f3817bebfa97816a0/java/src/main/java/ai/rapids/cudf/ColumnVector.java#L512